### PR TITLE
🏗 Attempt to fetch pinned chrome version up to 3 times per CI job

### PR DESCRIPTION
**Pull request by danielrozenberg**
Looks like we're stumbling on `429 Resource Exhausted` HTTP errors on occasion; this PR should mitigate that for now

e.g. before this PR (broken): https://app.circleci.com/pipelines/github/ampproject/amphtml/24690/workflows/ac9e9719-1277-4650-a6a7-ccdbde7729ec/jobs/506132?invite=true#step-113-10
with this PR (fixed): https://app.circleci.com/pipelines/github/ampproject/amphtml/24696/workflows/356fd8f7-8bc4-4a29-9678-6ba8f7a34020/jobs/506258/parallel-runs/1?filterBy=ALL&invite=true#step-113-10

### CODE REVIEW

File: .circleci/get_pinned_chrome_version.sh
Comment (rsimha) [inline .circleci/get_pinned_chrome_version.sh:53]: is `"null"` the only error that can be returned? I'm concerned that hard-coding an expected value could let other kinds of errors slip through.

File: .circleci/get_pinned_chrome_version.sh
Comment (rsimha) [inline .circleci/get_pinned_chrome_version.sh:50]: I wonder if we can simplify things by simply changing the `curl` command to `curl --retry 3`. With exponential backoff built in to `--retry`, this should virtually guarantee a result if the version history backend is indeed available. And if it isn't, it's okay for the job to fail, although it will be significantly more rare. WDYT?

**Review by rsimha — APPROVED** [review]
I've suggested what I think will be an easier and more maintainable fix. Will defer to you on which way to go. Approving to unblock.

File: .circleci/get_pinned_chrome_version.sh
Comment (danielrozenberg) [reply .circleci/get_pinned_chrome_version.sh]: `null` just means that the JSON object _didn't_ have an `"error"` field. Punted due to the other comment

File: .circleci/get_pinned_chrome_version.sh
Comment (danielrozenberg) [reply .circleci/get_pinned_chrome_version.sh]: Ha, this is so much better! The common error is `429` which is immediately retriable with `curl --retry`, so I reverted the change and just added `--retry 3` here. It also does exponential back-off on its own and respects `Retry-After` response headers

**Pull request merged**
